src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -1602,15 +1658,27 @@ static void lb_on_response_received_locked(grpc_exec_ctx *exec_ctx, void *arg,
             }
             grpc_grpclb_destroy_serverlist(serverlist);
           } else { /* new serverlist */
+            /* record the old status */
+            /* TODO(juanlishen): this assumes that num_servers never decreases
+             * to 0 once it's positive */
+            bool first_serverlist = glb_policy->serverlist == NULL;
```

Comment:
This code is running under the combiner (essentially a lock), so it's fine to clear `fallback_backend_addresses` before setting `serverlist`.  The important thing is that they're both done in the same place; it doesn't matter which happens first, because they're both done while holding the lock.

I think putting it in the `else` on line 1668 is fine.